lib/console_display.rb

```diff
@@ -1,22 +1,81 @@
 class ConsoleDisplay
 
   ROW_SEPARATOR = "\n -------------\n"
+  COLUMN_SEPARATOR = " | "
 
-  def initialize(game)
+  def initialize(game, console)
     @game = game 
+    @console = console
+  end
+
+  def start
+    show_start_screen
+    while !@game.over?
+      play_turn
+      show_board
+    end
+    display_outcome
+  end
+
+  def show_start_screen
+    show_greeting
+    show_board
+  end
+
+  def show_board
+    board = format_board
+    @console.show(board)
+  end
+
+  def play_round
+    play_empty_position
   end
 
   def format_board
     display = ROW_SEPARATOR
-    rows = create_rows
-    display += " | " + rows.flat_map { |row|
-      [row, ROW_SEPARATOR]}.join(" | ")
+    display += COLUMN_SEPARATOR + rows.flat_map { |row|
+      [row, ROW_SEPARATOR]}.join(COLUMN_SEPARATOR)
   end
 
   private
 
-  def create_rows
-    @game.board.cells.each_slice(3)
+  def rows
+    @game.board.row_content
+  end
+
+  def show_greeting
+    @console.display_greeting
+  end
+
+  def play_turn
+    @game.mark_position
+  end
+
+  def display_outcome
+    game_won?
```

Comment:
i would recommend moving the logic from `game_won?` and `game_drawn?` into here. These methods with a question marks have side effects (printing to screen), which isn't really expected from a reader's point of view.

So this becomes
```
if game.won?
  display_game_won_message
elsif game.drawn?
  display_game_drawn_message
end
```
The logic of what message to display is in one place, and not distributed across `game_won?` and `game_drawn?`.